docs/pipelines/index.mdx

```diff
@@ -3,6 +3,25 @@
 Feldera expresses incremental computations via `Pipelines`. A `Pipeline` is defined using SQL tables and views. Pipelines receive input data
 through input connectors. Pipelines can send the results computed by views to destinations via output connectors.
 
+## Pipeline architecture
+
+The diagram below illustrates the internal architecture of a Feldera pipeline:
+
+![Architecture of a Feldera Pipeline](pipeline_architecture.png)
+
+1. **Input connectors** ingest data from external sources such as message queues, databases, and data lakehouses.  A pipeline can have multiple
+  input connectors of different types, including several connectors feeding the same table.  An input connector buffers data in its
+  internal queue until the SQL engine is ready to accept more inputs.
+
+2. **Incremental SQL engine**. The Feldera SQL engine ingests buffered input data as changes to internal SQL tables. It performs
+  **Incremental View Maintenance (IVM)**, efficiently computing updates to SQL views based on input changes.
+
+3. **Output connectors** send changes to external data sinks. Like input connectors, a pipeline can have any number of
```

Comment:
There is a blog post about this too: https://www.feldera.com/blog/communicating-with-dbsp
Maybe we should update it to call this stuff "connectors" instead of "adapters"